### ISSUE TITLE
Unable to use remote btcpayserver LNBank as lightning connection (custom node) over lndhub for new btcpayserver

### ISSUE THREAD

**Issue by Amperstrand** [dennisreimann/btcpayserver-plugin-lnbank#51]
1. fresh install of a new btcpayserver via docker
2. click lightning (under wallets) to connect it to a lightning node
3. click use custom node
4. get lndhub credentials from an lnbank on a different btcpayserver (lnbank --> select the bank --> settings symbol --> show QR-code --> copy access url)
5. paste the lndhub credentials into the new btcpayserver:

```
lndhub://XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX:XXXXXXXXXXXX%3AXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@https//mybtcpayserver.com/plugins/lnbank/api/lndhub
```

6. click test connection
7. Get error message `Error while connecting to the API: Object reference not set to an instance of an object.`

I also tried using a different format for entering the lndhub credentials but get the same error:
type=lndhub;server=type=lndhub;server=[https://XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX:XXXXXXXXXXXX:XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@myotherbtcpayserver.com/plugins/lnbank/api/lndhub](https://XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX:XXXXXXXXXXXX%3AXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@myotherbtcpayserver.com/plugins/lnbank/api/lndhub)

Finally I made sure that the lndhub credentials were in fact working by importing them into Alby and making a transaction. The btcpayserver lnbank worked with alby communicating via LNDHub.

Lastly I tried my Alby account as the btcpayserver lightning backend:
1. Alby --> Connect your mobile wallet
2. LNDHub Credentials URI
3. copy `lndhub://XXXX:XXXX@https://ln.getalby.com/`
4. paste alby LNDHub credentials into btcpayserver
5. Connection to the Lightning node successful.

alby/bluewallet/zeus using LNBank over lndhub --> works
btcpayserver using Alby over lndhub --> works
btcpayserver using other btcpayserver LNBank over lndhub --> does not work (for me)

**Comment by dennisreimann** [comment]
What do the nginx and BTCPay Server logs on the destination look like when you use the Test Connection?

**Comment by Amperstrand** [comment]
nginx logs on btcpayserver with real lightning node when clicking test connection from btcpayserver over lndhub:

```
mybtcpayserver.com 1.2.3.4 - - [18/Nov/2023:20:28:20 +0000] "POST /plugins/lnbank/api/lndhubauth HTTP/1.1" 404 0 "-" "BTCPayServer.Lightning.LndHubClient"
```

nginx logs on btcpayserver with real lightning node when connecting from bluewallet (which works):
```
mybtcpayserver.com 2.2.2.2 - - [18/Nov/2023:20:32:48 +0000] "POST /plugins/lnbank/api/lndhub/auth?type=refresh_token HTTP/2.0" 200 118 "-" "okhttp/4.9.2"
mybtcpayserver.com 2.2.2.2 - - [18/Nov/2023:20:32:48 +0000] "GET /plugins/lnbank/api/lndhub/balance HTTP/2.0" 200 36 "-" "okhttp/4.9.2"
mybtcpayserver.com 2.2.2.2 - - [18/Nov/2023:20:32:48 +0000] "GET /plugins/lnbank/api/lndhub/gettxs?limit=10&offset=0 HTTP/2.0" 200 671 "-" "okhttp/4.9.2"
mybtcpayserver.com 2.2.2.2 - - [18/Nov/2023:20:32:48 +0000] "GET /plugins/lnbank/api/lndhub/getpending HTTP/2.0" 200 2 "-" "okhttp/4.9.2"
mybtcpayserver.com 2.2.2.2 - - [18/Nov/2023:20:32:48 +0000] "GET /plugins/lnbank/api/lndhub/getuserinvoices HTTP/2.0" 200 2 "-" "okhttp/4.9.2"
```

nginx logs on btcpayserver without real lightning node when clicking test connection to connect over lndhub:
```
myotherbtcpayserver.com 1.2.3.4 - - [18/Nov/2023:20:40:06 +0000] "POST /stores/3QdtD6PmZ5K1JupzExmrQcNhEP8haMcmTMQbBUEqGTAW/lightning/BTC/setup HTTP/2.0" 200 8173 "https://myotherbtcpayserver.com/stores/3QdtD6PmZ5K1JupzExmrQcNhEP8haMcmTMQbBUEqGTAW/lightning/BTC/setup" "myuseragent"
myotherbtcpayserver.com 1.2.3.4 - - [18/Nov/2023:20:40:06 +0000] "GET /stores/3QdtD6PmZ5K1JupzExmrQcNhEP8haMcmTMQbBUEqGTAW/lightning/BTC/_blazor?id=Z9xTgsPNL9EggPGY6Lej1w HTTP/1.1" 101 381 "-" "myuseragent"
myotherbtcpayserver.com 1.2.3.4 - - [18/Nov/2023:20:40:06 +0000] "POST /stores/3QdtD6PmZ5K1JupzExmrQcNhEP8haMcmTMQbBUEqGTAW/lightning/BTC/_blazor/disconnect HTTP/2.0" 200 0 "https://myotherbtcpayserver.com/stores/3QdtD6PmZ5K1JupzExmrQcNhEP8haMcmTMQbBUEqGTAW/lightning/BTC/setup" "myuseragent"
myotherbtcpayserver.com 1.2.3.4 - - [18/Nov/2023:20:40:06 +0000] "GET /stores/3QdtD6PmZ5K1JupzExmrQcNhEP8haMcmTMQbBUEqGTAW/lightning/BTC/_blazor/initializers HTTP/2.0" 200 28 "https://myotherbtcpayserver.com/stores/3QdtD6PmZ5K1JupzExmrQcNhEP8haMcmTMQbBUEqGTAW/lightning/BTC/setup" "myuseragent"
myotherbtcpayserver.com 1.2.3.4 - - [18/Nov/2023:20:40:06 +0000] "POST /stores/3QdtD6PmZ5K1JupzExmrQcNhEP8haMcmTMQbBUEqGTAW/lightning/BTC/_blazor/negotiate?negotiateVersion=1 HTTP/2.0" 200 316 "https://myotherbtcpayserver.com/stores/3QdtD6PmZ5K1JupzExmrQcNhEP8haMcmTMQbBUEqGTAW/lightning/BTC/setup" "myuseragent"
```

This is an error in the btcpayserver logs of myotherbtcpayserver.com which I will look further into:
```
info: Configuration:  Now listening on: http://0.0.0.0:49392
fail: Microsoft.AspNetCore.Server.Kestrel: Connection id "0HMV88TEGL19U", Request id "0HMV88TEGL19U:00000002": An unhandled exception was thrown by the application.
System.Net.Sockets.SocketException (99): Cannot assign requested address
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.CreateException(SocketError error, Boolean forAsyncThrow)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ConnectAsync(Socket socket)
   at System.Net.Sockets.Socket.ConnectAsync(EndPoint remoteEP, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.ConnectAsync(EndPoint remoteEP)
   at BTCPayServer.Lightning.CLightning.CLightningClient.Connect()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at BTCPayServer.Lightning.CLightning.CLightningClient.Connect()
   at BTCPayServer.Lightning.CLightning.CLightningClient.SendCommandAsync[T](String command, Object[] parameters, Boolean noReturn, Boolean isArray, CancellationToken cancellation)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at BTCPayServer.Lightning.CLightning.CLightningClient.SendCommandAsync[T](String command, Object[] parameters, Boolean noReturn, Boolean isArray, CancellationToken cancellation)
   at BTCPayServer.Lightning.CLightning.CLightningClient.ListFundsAsync(CancellationToken cancellation)
   at BTCPayServer.Lightning.CLightning.CLightningClient.BTCPayServer.Lightning.ILightningClient.GetBalance(CancellationToken cancellation)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at BTCPayServer.Lightning.CLightning.CLightningClient.BTCPayServer.Lightning.ILightningClient.GetBalance(CancellationToken cancellation)
   at BTCPayServer.Controllers.Greenfield.GreenfieldLightningNodeApiController.GetBalance(String cryptoCode, CancellationToken cancellationToken) in /source/BTCPayServer/Controllers/GreenField/GreenfieldLightningNodeApiController.cs:line 62
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at BTCPayServer.Controllers.Greenfield.GreenfieldLightningNodeApiController.GetBalance(String cryptoCode, CancellationToken cancellationToken)
   at BTCPayServer.Controllers.Greenfield.GreenfieldInternalLightningNodeApiController.GetBalance(String cryptoCode, CancellationToken cancellationToken) in /source/BTCPayServer/Controllers/GreenField/GreenfieldLightningNodeApiController.Internal.cs:line 53
   at BTCPayServer.Controllers.Greenfield.LocalBTCPayServerClient.GetLightningNodeBalance(String cryptoCode, CancellationToken token) in /source/BTCPayServer/Controllers/GreenField/LocalBTCPayServerClient.cs:line 478
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at BTCPayServer.Controllers.Greenfield.LocalBTCPayServerClient.GetLightningNodeBalance(String cryptoCode, CancellationToken token)
   at BTCPayServer.Plugins.LNbank.Services.BTCPayService.GetLightningNodeBalance(CancellationToken cancellationToken) in /build/BTCPayServer.Plugins.LNbank/Services/BTCPayService.cs:line 61
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at BTCPayServer.Plugins.LNbank.Services.BTCPayService.GetLightningNodeBalance(CancellationToken cancellationToken)
   at BTCPayServer.Plugins.LNbank.Pages.Wallets.IndexModel.OnGetAsync(String walletId) in /build/BTCPayServer.Plugins.LNbank/Pages/Wallets/Index.cshtml.cs:line 61
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at BTCPayServer.Plugins.LNbank.Pages.Wallets.IndexModel.OnGetAsync(String walletId)
   at lambda_method1246(Closure , Object , Object[] )
```

Also worth noting is that myotherbtcpayserver is deployed via docker, is connected to core-lightning, but core-lightning has no channels since it is intended to use via lnbank.

**Comment by dennisreimann** [comment]
The path `/plugins/lnbank/api/lndhubauth` is incorrect and needs to be `/plugins/lnbank/api/lndhub/auth`. Can you try adjusting the path and add a trailing slash to it?

**Comment by Amperstrand** [comment]
If I add a trailing auth to the credentials that I copied from the first btcpayserver with the real lightning node and paste the following into the field before I click on test connection:

```
lndhub://XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX:XXXXXXXXXXXX%3AXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@https//mybtcpayserver.com/plugins/lnbank/api/lndhub/auth
```

Then I get the following showing up in the nginx logs of the first btcpayserver:

```
mybtcpayserver.com 1.2.3.4 - - [19/Nov/2023:15:46:11 +0000] "POST /plugins/lnbank/api/lndhub/auth/auth HTTP/1.1" 404 0 "-" "BTCPayServer.Lightning.LndHubClient"
```

It looks like the lndhub client is adding /auth to the URL.

The credentials for lndhub are copied from the first btcpayserver. So if the /auth should be part of the URL, then that would need changeing.

It is also strange that the same URL works with getalby.

**Comment by Amperstrand** [comment]
Adding a trailing /

so that it is:

```
lndhub://XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX:XXXXXXXXXXXX%3AXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@https//mybtcpayserver.com/plugins/lnbank/api/lndhub/auth/
```

Gives me this error instead:

```
Error while connecting to the API: Bad auth
```

The credentials are correct and tested with Alby.

Now nginx logs on btcpayserver are:

```
mybtcpayserver.com 1.2.3.4 - - [19/Nov/2023:15:56:51 +0000] "POST /plugins/lnbank/api/lndhub/auth HTTP/1.1" 200 44 "-" "BTCPayServer.Lightning.LndHubClient"
```

**Comment by dennisreimann** [comment]
Fixed in LNbank v1.8.8 — thanks for reporting!
